Valansch/RedMew
# Updated donor features

Features:
- Team buffs for donators with tier 2 or higher Patreon status
- Tier 2 (coal): +10% team hand mining bonus per online donator, up to a maximum of 10 online donators
- Tier 3 (solid fuel): +10% team hand crafting bonus per online donator, up to a max of 10 online donators
- Tier 4 (rocket fuel): +10% team run speed per online donator, up to a max of 10 online donators
- Tier 5 (nuclear fuel): +5 inventory slots for all. Up to max of 30 extra. Does not reduce down when donator logs off.

To do:
- How do I call a function on_load? The commend on line 344 didn't work. I want to check the list of applied bonuses when the server loads after a crash or stop to see who is online and apply the correct buffs.
- Update donator_tiers[X].max to 10, 10, 10, 10 and 6. The low values are only there for testing.

Tests to perform before merging:
- Use this command to check the current modifiers: /c local force = game.player.force game.print("Mining modifier: "..force.manual_mining_speed_modifier.." Crafting modifier: "..game.player.force.manual_crafting_speed_modifier.." Run modifier: "..game.player.force.character_running_speed_modifier.." Extra slots: "..force.character_inventory_slots_bonus)
- Check it doesn't interact badly with diggy mining speed
- Confirm that it works correctly with the market mining and running boosts
- Ensure that the max values work correctly. For now the limit has been set to max 2 donators per level. This will change back once tested.
